src/Parameters/BaseParameters.php

```diff
@@ -23,18 +23,78 @@
  */
 abstract class BaseParameters
 {
-    /**
-     * @param $array
-     *
-     * @return string
-     */
-    protected function buildHTTPQuery($array)
+    public function __call(string $name, array $arguments)
+    {
+        if (strpos($name, 'get') === 0) {
+            return $this->getter(\lcfirst(substr($name, 3)));
+        } elseif (strpos($name, 'is') === 0) {
+            return $this->booleanGetter(\lcfirst(substr($name, 2)));
+        } elseif (strpos($name, 'set') === 0) {
+            return $this->setter(lcfirst(substr($name, 3)), $arguments);
+        } else {
+            throw new \BadFunctionCallException($name . ' does not exist');
+        }
+    }
+
+    protected function getter($name)
+    {
+        if (property_exists($this, $name)) {
+            return $this->$name;
+        } else {
+            throw new \BadFunctionCallException($name . ' is not a valid property');
+        }
+    }
+
+    protected function booleanGetter($name)
+    {
+        $value = $this->getter($name);
+
+        if (!\is_bool($this->$name) && $this->$name !== null) {
+            throw new \BadFunctionCallException($name . ' is not a boolean property');
+        }
+
+        return $value;
+    }
+
+    protected function setter($name, $arguments)
     {
-        return http_build_query(array_filter($array));
+        if (!property_exists($this, $name)) {
+            throw new \BadFunctionCallException($name . ' is not a valid property');
+        }
+
+        $this->$name = $arguments[0];
+
+        return $this;
+    }
+
+    protected function getProperties(): array
```

Comment:
what about adding an empty array like `$ignoreProperties = [];` instead of the getProperties method. That would save the getRawXml and setRawXml methods in the src/Parameters/SetConfigXMLParameters.php

Reply:
What would be the benefit? Do you think it will improve performance, or make the ignore process more clear?

Reply:
I think it could be more clear than the underscore convention and would allow the usage of the getters and setters (keeps the code more DRY)

Reply:
I agree that this would be more clear, but what do you mean by "allow the usage of the getters and setters".

Reply:
> I agree that this would be more clear, but what do you mean by "allow the usage of the getters and setters".

Should have looked at SetConfigXMLParameters, before asking... Sorry